models/stg_apple_store__sales_subscription_events.sql

```diff
@@ -30,21 +30,38 @@ final as (
+        cast(_fivetran_id as {{ dbt.type_string() }}) as _fivetran_id,
+        cast(vendor_number as {{ dbt.type_int() }}) as vendor_number,
+        cast(app_apple_id as {{ dbt.type_int() }}) as app_apple_id,
         cast(app_name as {{ dbt.type_string() }}) as app_name,
-        cast(subscription_name as {{ dbt.type_string() }}) as subscription_name,
         cast(event as {{ dbt.type_string() }}) as event,
+        cast(subscription_name as {{ dbt.type_string() }}) as subscription_name,
+        cast(subscription_apple_id as {{ dbt.type_int() }}) as subscription_apple_id,
+        cast(subscription_group_id as {{ dbt.type_int() }}) as subscription_group_id,
+        cast(standard_subscription_duration as {{ dbt.type_string() }}) as standard_subscription_duration,
+        cast(subscription_offer_type as {{ dbt.type_string() }}) as subscription_offer_type,
+        cast(subscription_offer_duration as {{ dbt.type_string() }}) as subscription_offer_duration,
+        cast(marketing_opt_in as {{ dbt.type_string() }}) as marketing_opt_in,
+        cast(marketing_opt_in_duration as {{ dbt.type_string() }}) as marketing_opt_in_duration,
+        cast(preserved_pricing as {{ dbt.type_string() }}) as preserved_pricing,
+        cast(proceeds_reason as {{ dbt.type_string() }}) as proceeds_reason,
+        cast(promotional_offer_name as {{ dbt.type_string() }}) as promotional_offer_name,
+        cast(promotional_offer_id as {{ dbt.type_string() }}) as promotional_offer_id,
+        cast(consecutive_paid_periods as {{ dbt.type_int() }}) as consecutive_paid_periods,
+        cast(original_start_date as date) as original_start_date,
+        cast(device as {{ dbt.type_string() }}) as device,
+        cast(client as {{ dbt.type_string() }}) as client,
+        cast(state as {{ dbt.type_string() }}) as state,
         cast(country as {{ dbt.type_string() }}) as country,
-        cast(case
-            when replace(state, ' ', '') = '' then cast(null as {{ dbt.type_string() }}) else state
-        end as {{ dbt.type_string() }}) as state,
-        cast(case 
-            when lower(device) like 'ipod%' then 'iPod' else device
-        end as {{ dbt.type_string() }}) as device,
-        sum(cast(quantity as {{ dbt.type_bigint() }})) as quantity
+        cast(previous_subscription_name as {{ dbt.type_string() }}) as previous_subscription_name,
+        cast(previous_subscription_apple_id as {{ dbt.type_int() }}) as previous_subscription_apple_id,
+        cast(days_before_canceling as {{ dbt.type_int() }}) as days_before_canceling,
+        cast(cancellation_reason as {{ dbt.type_string() }}) as cancellation_reason,
+        cast(days_canceled as {{ dbt.type_int() }}) as days_canceled,
+        cast(quantity as {{ dbt.type_int() }}) as quantity,
+        cast(paid_service_days_recovered as {{ dbt.type_int() }}) as paid_service_days_recovered
     from fields
-    {{ dbt_utils.group_by(9) }}
```

Comment:
Did the grain of this model change? Is there a reason we no longer need this group by?

Reply:
Yup I changed the grain-- I thought it was non-standard to have aggregations in the staging model. I understand why it was done (to only filter to relevant columns and aggregating `quantity` to ensure all values are captured), but I figured it made more sense to have everything in staging, then keep the aggregations in the transforms like how we have it elsewhere.